packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -1122,4 +1125,351 @@ describe('KeyringController', () => {
+            ],
+          },
+          to: [
+            {
+              name: 'Bob',
+              wallets: [
+                '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+                '0xB0BdaBea57B0BDABeA57b0bdABEA57b0BDabEa57',
+                '0xB0B0b0b0b0b0B000000000000000000000000000',
+              ],
+            },
+          ],
+        },
+        primaryType: 'Mail' as const,
+        types: {
+          EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+          ],
+          Group: [
+            { name: 'name', type: 'string' },
+            { name: 'members', type: 'Person[]' },
+          ],
+          Mail: [
+            { name: 'from', type: 'Person' },
+            { name: 'to', type: 'Person[]' },
+            { name: 'contents', type: 'string' },
+          ],
+          Person: [
+            { name: 'name', type: 'string' },
+            { name: 'wallets', type: 'address[]' },
+          ],
+        },
+      };
+
+      const confirmSignatureStub = sinon.stub(ledgerKeyring, 'signTypedData');
+
+      confirmSignatureStub.resolves(
+        '0xafb6e247b1c490e284053c87ab5f6b59e219d51f743f7a4d83e400782bc7e4b9479a268e0e0acd4de3f1e28e4fac2a6b32a4195e8dfa9d19147abe8807aa6f6400',
+      );
+
+      const account = await ledgerKeyring.getDefaultAccount();
+      const signature = await keyringController.signTypedMessage(
+        { data: JSON.stringify(msgParams), from: account },
+        SignTypedDataVersion.V4,
+      );
+      const recovered = recoverTypedSignature({
+        data: msgParams,
+        signature,
+        version: SignTypedDataVersion.V4,
+      });
+      expect(account).toBe(recovered);
+    });
+
+    it('should forget Ledger Keyring', async () => {
+      const accounts = await ledgerKeyring.getAccounts();
+      expect(accounts).toHaveLength(1);
+
+      await keyringController.forgetLedger();
+      expect(keyringController.state.keyrings[1].accounts).toHaveLength(0);
+    });
+
+    it('should return Ledger Keyring for corresponding account', async () => {
+      const ledgerAccount = '0xe908e4378431418759b4f87b4bf7966e8aaa5cf2';
+
+      const keyring = await keyringController.getAccountKeyringType(
+        ledgerAccount,
+      );
+
+      expect(keyring).toBe(KeyringTypes.ledger);
+    });
+
+    it('should connect to ledger hardware', async () => {
+      const setTransportStub = sinon.stub(ledgerKeyring, 'setTransport');
+      setTransportStub.resolves();
+
+      const getAppAndVersionStub = sinon.stub(
+        ledgerKeyring,
+        'getAppAndVersion',
+      );
+
+      const mockedAppVersionValue = {
+        appName: 'mockApp',
+        version: 'mockVersion',
+      };
+      getAppAndVersionStub.resolves(mockedAppVersionValue);
+
+      const mockTransport = {} as Transport;
```

Comment:
Is there any way we can provide a real transport object here instead of typecasting?

Reply:
I have implemented a mock version of the transport in a separate file.